### PR TITLE
 Add `upgrade` action to the pyenv_pip resource

### PR DESCRIPTION
**Pull request by lukeseawalker**
# Description

 Package will be updated to latest if version is not provided.

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [ X ] All tests pass. See TESTING.md for details.
- [ X ] New functionality includes testing.
- [ X ] New functionality has been documented in the README if applicable.

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
@lukeseawalker can you please address the two issues mentioned by kitchen-porter?

**Comment by lukeseawalker** [comment]
@ramereth I'm not sure about the "Release Label Validator" check

File: resources/pip.rb [hunk 1]
```diff
@@ -98,8 +98,10 @@ def require_install?
     end
 
     unless new_resource.version
-      Chef::Log.debug("already installed: #{new_resource.package_name} #{current_version}")
-      return false
+      unless new_resource.options.split(/\s+/).include? '--upgrade'
+        Chef::Log.debug("already installed: #{new_resource.package_name} #{current_version}")
+        return false
+      end
```
Comment (ramereth) [inline resources/pip.rb:104]: Looking at this more closely, I feel a better approach might be to create a new `:upgrade` action which can properly deal with this logic a little better.

**Review by ramereth — CHANGES_REQUESTED** [review]

**Comment by ramereth** [comment]
@lukeseawalker can you please update the documentation to reflect this new action?

**Review by ramereth — APPROVED** [review]

**Pull request merged**